src/test/java/org/commcare/formplayer/tests/MediaCaptureTest.kt

```diff
@@ -0,0 +1,109 @@
+package org.commcare.formplayer.tests
+
+import org.commcare.formplayer.beans.FormEntryResponseBean
+import org.commcare.formplayer.beans.NewFormResponse
+import org.commcare.formplayer.utils.FileUtils
+import org.javarosa.core.services.locale.Localization
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.junit.jupiter.api.fail
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest
+import org.springframework.mock.web.MockMultipartFile
+import org.springframework.test.util.AssertionErrors.assertFalse
+import org.springframework.test.util.AssertionErrors.assertTrue
+import java.io.File
+
+@WebMvcTest
+class MediaCaptureTest : BaseTestClass() {
+
+    companion object {
+        const val USERNAME = "test"
+        const val DOMAIN = "test"
+        const val APP_ID = "10a706429116a3e55f1d1302cd3db69f"
+        const val IMAGE_CAPTURE_INDEX = 21
+    }
+
+    @Test
+    fun testImageCapture_fileSaveAndReplace() {
+        val formResponse = startImageCaptureForm()
+        val responseBean: FormEntryResponseBean
+        try {
+            responseBean = saveImage(formResponse, "media/valid_image.jpg", "valid_image.jpg")
+        } catch (e: Exception) {
+            fail("Unable to save a valid file due to " + e.message)
+        }
```

Comment:
why not just let the exception get raised?

Reply:
I was trying to be more assertive here with the primary intent of this test being testing the file save.